src/library/scala/collection/mutable/FlatHashTable.scala

```diff
@@ -415,20 +418,7 @@ private[collection] object FlatHashTable {
     // so that:
     protected final def sizeMapBucketSize = 1 << sizeMapBucketBitSize
 
-    protected final def improve(hcode: Int, seed: Int) = {
-      //var h: Int = hcode + ~(hcode << 9)
-      //h = h ^ (h >>> 14)
-      //h = h + (h << 4)
-      //h ^ (h >>> 10)
-
-      val improved= scala.util.hashing.byteswap32(hcode)
-
-      // for the remainder, see SI-5293
-      // to ensure that different bits are used for different hash tables, we have to rotate based on the seed
-      val rotation = seed % 32
-      val rotated = (improved >>> rotation) | (improved << (32 - rotation))
-      rotated
-    }
+    protected final def improve(hcode: Int, seed: Int) = rotr(byteswap32(hcode), seed & ((1 << 5) - 1))
```

Comment:
Can you just use `& 0x1F` or `& 31`?  That `((1 << 5) - 1)` happens to be the same isn't going to change, and I think the mask-like nature of it is at least as clear at a glance since it takes a couple seconds to wade through the parens and stuff.  (And it hashtable also.)  Otherwise, looks good.